src/game_state.rs

```diff
@@ -3,14 +3,38 @@ use rand;
 use geometry::{Position, Size};
 use models::World;
 
+/// This is a message that will be drawn to the screen. When it's shown on the screen the game
+/// will be waiting for user input
+pub struct Message {
+    pub title: &'static str,
+    pub subtitle: &'static str,
+    pub should_show: bool
+}
+
+/// The Message to show when the game starts
+const WELCOME_MESSAGE: Message = Message {
+    title: "Welcome to Rocket!",
+    subtitle: "Press any key to start",
+    should_show: true
+};
+
+/// The Message to show when the game is over
+const GAMEOVER_MESSAGE: Message = Message {
+    title: "Game Over",
+    subtitle: "Press any key to restart",
+    should_show: true
+};
+
 /// The data structure that contains the state of the game
 pub struct GameState {
     /// The world contains everything that needs to be drawn
     pub world: World,
     /// The current difficulty - the enemies will speed up over time
     pub difficulty: f64,
+    /// Information about the Message to draw on the screen
+    pub message: Message,
```

Comment:
I would expect this to be an `Option<Message>`

Reply:
Ah, being an `Option<Message>` is much better - `should_show` definitely was a hack!